### PR TITLE
Add partitions test to GKE

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Update GKE terraform templates for partitions tests
  - we create firewall rules that allow nodes and pods from each cluster to reach the nodes of the other.
- Update server podsercuritypolicy to allow hostports on specific ports if `server.exportGossipAndRPCPorts` is set to `true`

How I've tested this PR:
- ran tests: https://app.circleci.com/pipelines/github/hashicorp/consul-k8s/3926/workflows/1aeed2ff-40b4-4a7a-99cb-f84f751048a0

How I expect reviewers to test this PR:
👀 

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

**Comment by thisisnotashwin** [comment]
The PR is ready. Waiting for it to go 🟢 on CI before marking it ready for review.

File: .circleci/config.yml
Comment (thisisnotashwin) [inline .circleci/config.yml:975]: These will be removed before merging!

File: charts/consul/templates/server-podsecuritypolicy.yaml
Comment (thisisnotashwin) [inline charts/consul/templates/server-podsecuritypolicy.yaml:35]: Im not sure what happens here if the value of the `serflan.port` is greater than 8302. Does the next entry then clobber it?

File: charts/consul/test/terraform/gke/variables.tf
Comment (thisisnotashwin) [inline charts/consul/test/terraform/gke/variables.tf:29]: Always a good idea to steal from @ishustava

File: charts/consul/templates/server-podsecuritypolicy.yaml
Comment (thisisnotashwin) [reply charts/consul/templates/server-podsecuritypolicy.yaml:35]: This definitely works with the default values. A touch worried about non-default ones.

Comment (ishustava) [reply charts/consul/templates/server-podsecuritypolicy.yaml:35]: I don't think the next entry would clobber it. I think each array item is an independent port range.

File: charts/consul/test/terraform/gke/main.tf
Comment (ishustava) [inline charts/consul/test/terraform/gke/main.tf:41]: need to update description I think to reflect what it does
```suggestion
  description = "Creates firewall rule allowing traffic from nodes and pods of the ${random_id.suffix[count.index == 0 ? 1 : 0].dec} Kubernetes cluster."
```

File: charts/consul/test/terraform/gke/variables.tf
Comment (ishustava) [reply charts/consul/test/terraform/gke/variables.tf:29]: Great artists steal... 😄

**Review by ishustava — APPROVED** [review]
Looks great! A couple of minor comments, but non-blocking.

**Review by ndhanushkodi — APPROVED** [review]
Looks great Ashwin!!

**Pull request merged**
